packages/runtime/src/public/createApp.ts

```diff
@@ -0,0 +1,135 @@
+// Note: We should be careful of what we import in this module, we don't want to load the entire runtime here
+// until createApp is called, since we set some globals here, which will be read by other runtime modules
+import '@mobilestack-xyz/runtime/src/missingGlobals'
+import Config from 'react-native-config'
+import DeviceInfo from 'react-native-device-info'
+import { ToggleableOnboardingFeatures } from '../onboarding/types'
+
+// This will evolve. We should be mindful of breaking changes.
+// This structure should scale well as we add more features
+// and makes it clear what's core configuration vs optional features.
+//
+// Guidelines:
+// - We should only have a few core configuration options, and the rest should be optional features and/or with default values
+// - We should only add new configuration options that we want to support long term, and not just for a specific app
+// - Configuration options should be well documented and have clear purposes
+// - Breaking changes to configuration should be avoided when possible
+// - Configuration should be type-safe. In some cases we can consider runtime validation.
+interface PublicAppConfig {
+  registryName: string
+  displayName: string
+  deepLinkUrlScheme: string
+
+  // Platform specific configuration
+  ios?: {
+    appStoreId?: string
+  }
+
+  // Theme configuration
+  themes?: {
+    // Rough example of what we might want to support
+    default: {
+      // To adjust status bar style, keyboard appearance, etc
+      isDark?: boolean
+      colors?: {
+        // Core brand colors
+        primary?: string
+        secondary?: string
+        background?: string
+
+        // Semantic colors
+        error?: string
+        success?: string
+        warning?: string
+
+        // Text colors
+        text?: {
+          primary?: string
+          secondary?: string
+          disabled?: string
+        }
+      }
+    }
+  }
+
+  // Optional features/capabilities
+  features?: {
+    sentry?: {
+      clientUrl: string
+    }
+    // TODO: what's the marketing name for this?
+    cloudBackup?: boolean
```

Comment:
So for cloud backup we have a few environment variables and secrets that we use.
`WEB3AUTH_CLIENT_ID` and `AUTH0_CLIENT_ID` which currently are in secrets.json and `SMS_RETRIEVER_APP_SIGNATURE` which lives in the .env files. Where do those live now for this plan?

Reply:
Yes I was just talking about this with @satish-ravi, `WEB3AUTH_CLIENT_ID` and `AUTH0_CLIENT_ID` will be directly included into the library. They aren't real secrets and are present in the current app bundle for anyone looking at it.
As for `SMS_RETRIEVER_APP_SIGNATURE` so isn't actually consumed in the app, it's needed by backend services and was stored in our .env as a reference.